protobuf/v1/snapshot.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+
+import "nexus.proto";
+package mayastor.v1;
+
+service SnapshotRpc {
+    // Common snapshot related methods.
+    rpc CreateReplicaSnapshot(CreateReplicaSnapshotRequest) returns (CreateReplicaSnapshotResponse) {}
+    rpc CreateNexusSnapshot(NexusCreateSnapshotRequest) returns (NexusCreateSnapshotResponse) {}
+    rpc ListSnapshot(ListSnapshotsRequest) returns (ListSnapshotsResponse) {}
+    rpc DestroySnapshot(DestroySnapshotRequest) returns (google.protobuf.Empty) {}
+}
+
+/// Parameter details of a single snapshot
+message SnapshotInfo {
+    string snapshot_uuid = 1; // uuid of the snapshot
+    string snapshot_name = 2; // name of the snapshot
+    uint64 snapshot_size = 3; // amount of bytes referenced by snapshot
+    uint64 num_clones = 4; // number of clones created from this snapshot
+    google.protobuf.Timestamp timestamp = 5; // snapshot creation_timestamp
+    string source_uuid = 6; // uuid of the source from which snapshot is created (nexus / replica)
+    uint64 source_size = 7; // amount of bytes referenced by source
+    string pool_uuid = 8; // pool UUID in which snapshot is present
```

Comment:
can we get the pool_id as well? (ie the pool name)

Reply:
> can we get the pool_id as well? (ie the pool name)

Updated the interface and added `pool_name` as well in the response.